src/generators/baseGenerator.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  AnySchemaLike,
+  FromSchemaLike,
+  createSchemaValidator,
+  toJsonSchema,
+} from "@/internals/helpers/schema.js";
+import { GenerateOptions, LLMError } from "@/llms/base.js";
+import { ChatLLM } from "@/llms/chat.js";
+import { BaseMessage, Role } from "@/llms/primitives/message.js";
+import { Retryable } from "@/internals/helpers/retryable.js";
+import { PromptTemplate } from "@/template.js";
+import { SchemaObject } from "ajv";
+
+export interface GenerateSchemaInput<T> {
+  maxRetries?: number;
+  options?: T;
+}
+
+export abstract class BaseGenerator<TGenerateOptions extends GenerateOptions = GenerateOptions> {
+  protected abstract template: PromptTemplate<"schema">;
+  protected errorTemplate = new PromptTemplate<"expected" | "received" | "errors">({
+    variables: ["errors", "expected", "received"],
+    template: `Generated response does not match the expected schema!
+Validation Errors: "{{errors}}"`,
+  });
```

Comment:
The `PromptTemplate` implementation has changed (use `schema` instead of `variables`).